File: fornituras-api/src/main/java/com/numobiz/solutions/fornituras/modules/warehouses/entity/Warehouse.java
package com.numobiz.solutions.fornituras.modules.warehouses.entity;

import com.numobiz.solutions.fornituras.common.entity.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;

/**
 * Almacén: entidad operativa (no catálogo plano). Ubicación física de resguardo de fornituras,
 * con clave de negocio, clasificación, ubicación, responsable y cupo.
 *
 * <p>Las referencias a municipio y responsable se guardan como id (FK en BD) en lugar de
 * asociaciones JPA para mantener el módulo desacoplado de {@code municipios} y {@code users};
 * la integridad referencial la garantiza la migración. Sin PII de elementos; los campos
 * sensibles (dirección, geolocalización, responsable, contacto) se protegen por autorización.
 */
@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "warehouse")
public class Warehouse extends BaseEntity {

	/** Clave de negocio estable (p. ej. "ALM-01"); referenciada por traslados y etiquetas. */
	@Column(nullable = false, unique = true, length = 40)
	private String codigo;

	@Column(nullable = false, length = 120)
	private String nombre;

	/** Nombre normalizado (trim + colapso + casefold + sin acentos) para unicidad. */
	@Column(name = "nombre_normalizado", nullable = false, unique = true, length = 120)
	private String nombreNormalizado;

	@Enumerated(EnumType.STRING)
	@Column(nullable = false, length = 20)
	private WarehouseType tipo;

	@Column(name = "municipio_id")
	private Long municipioId;

	@Column(length = 255)
	private String direccion;

	@Column(length = 10)
	private String cp;

	@Column(precision = 9, scale = 6)
	private BigDecimal latitud;

	@Column(precision = 9, scale = 6)
	private BigDecimal longitud;

	@Column(name = "responsable_id")
	private Long responsableId;

	@Column(length = 30)
	private String telefono;

	@Column(name = "email_contacto", length = 255)
	private String emailContacto;

	private Integer capacidad;

	@Column(length = 500)
	private String observaciones;

	@Column(nullable = false)
	private boolean active = true;
}

File: fornituras-api-dotnet/src/Fornituras.Api/Data/Entities/Warehouse.cs
namespace Fornituras.Api.Data.Entities;

/// <summary>
/// Almacén: ubicación física de resguardo de fornituras.
/// </summary>
public class Warehouse : BaseEntity
{
    public string Codigo { get; set; } = string.Empty;

    public string Nombre { get; set; } = string.Empty;

    public string NombreNormalizado { get; set; } = string.Empty;

    /// <summary>FK a un valor del catálogo TIPO_ALMACEN.</summary>
    public long TipoItemId { get; set; }

    public string? Municipio { get; set; }

    public string? Estado { get; set; }

    public string? Direccion { get; set; }

    public string? Cp { get; set; }

    public decimal? Latitud { get; set; }

    public decimal? Longitud { get; set; }

    public long? ResponsableId { get; set; }

    public string? Telefono { get; set; }

    public string? EmailContacto { get; set; }

    public int? Capacidad { get; set; }

    public string? Observaciones { get; set; }

    public bool Active { get; set; } = true;
}
